### PR TITLE
fix(Core): Save bonus talent points to DB

### PR DESCRIPTION
**Pull request by ZhengPeiRu21**
## Changes Proposed:
-  Save player bonus talent points to the DB

## Issues Addressed:

- Allows bonus talents to be preserved through relogs
- Avoids forced talent reset on relog if bonus talent points had been granted
- Closes https://github.com/ZhengPeiRu21/mod-challenge-modes/issues/12

## Tests Performed:

- Tested in game, works

## How to Test the Changes:

1. Add bonus talents to player using SetBonusTalents (challenge modes module can be configured to do this)
2. Log out, confirm talents do not reset and bonus points are still available

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/Player/PlayerStorage.cpp [hunk 1]
```diff
@@ -5473,7 +5473,10 @@ bool Player::LoadFromDB(ObjectGuid playerGuid, CharacterDatabaseQueryHolder cons
     _LoadMonthlyQuestStatus(holder.GetPreparedResult(PLAYER_LOGIN_QUERY_LOAD_MONTHLY_QUEST_STATUS));
     _LoadRandomBGStatus(holder.GetPreparedResult(PLAYER_LOGIN_QUERY_LOAD_RANDOM_BG));
 
-    // after spell and quest load
+    // Extra Bonus Talent Points
+    m_extraBonusTalentCount = fields[73].Get<uint8>();
+
+    // after spell, bonus talents and quest load
```
Comment (heyitsbench) [inline src/server/game/Entities/Player/PlayerStorage.cpp:5479]: [Oxford comma](https://www.grammarly.com/blog/what-is-the-oxford-comma-and-why-do-people-care-so-much-about-it/)? :3

**Comment by StraysFromPath** [comment]
I have tested this PR and it fixes the issue. I tested some cases like adding 100 talent points, relogging, server restart, etc.

Talent point reset is very detrimental to custom servers and also challenge modes.

**Review by StraysFromPath — APPROVED** [review]
I have tested this PR on my docker build and it fixes the issue of custom talents resetting on login. I tested some cases like adding 100 talent points, relogging, server restart, etc.

Talent point reset is very detrimental to custom servers and also challenge modes.

**Comment by acidmanifesto** [comment]
please update the stream

**Review by acidmanifesto — APPROVED** [review]

**Comment by Maelthyrr** [comment]
@ZhengPeiRu21 could you update the stream?

**Comment by ZhengPeiRu21** [comment]
Updated.

**Review by Maelthyrr — APPROVED** [review]

**Pull request merged**